flask_rest_api/spec/__init__.py

```diff
@@ -87,24 +87,29 @@ def _register_doc_blueprint(self, app):
                 url_prefix=_add_leading_slash(api_url),
                 template_folder='./templates',
             )
-            # Serve json spec at 'url_prefix/openapi.json' by default
-            json_path = app.config.get(
-                'OPENAPI_JSON_PATH', 'openapi.json')
-            blueprint.add_url_rule(
-                _add_leading_slash(json_path),
-                endpoint='openapi_json',
-                view_func=self._openapi_json)
+            self._register_openapi_json_rule(app, blueprint)
             self._register_redoc_rule(app, blueprint)
             self._register_swagger_ui_rule(app, blueprint)
             app.register_blueprint(blueprint)
 
-    def _openapi_json(self):
-        """Serve JSON spec file"""
-        # We don't use Flask.jsonify here as it would sort the keys
-        # alphabetically while we want to preserve the order.
-        return current_app.response_class(
-            json.dumps(self.spec.to_dict(), indent=2),
-            mimetype='application/json')
+    def _register_openapi_json_rule(self, app, blueprint):
+        """Serve json spec file"""
+        json_path = app.config.get('OPENAPI_JSON_PATH', 'openapi.json')
+
```

Comment:
Similar to my other comment, I think you could use `current_app` instead of a closure with `app`. Except that here you may need to use: `current_app._get_current_object()` to access `self.specs`

Reply:
Yes, perhaps I could store the spec object in `app.extensions['flask-rest-api']['spec']` and access it from there.

I don't have a strong feeling about it. Any downside with the closure approach?

Reply:
I don't have a strong argument against the closure approach. I just find a bit confusing having references to both `current_app` and `app` in the same function. I would pick one way to reference the application instance. Or maybe I'm missing something (?)

Reply:
Yeah, I agree it looks strange to use both. As I mentioned in the comment, I was using `current_app` here on the wrong assumption that `current_app.response_class` might be resolved at function definition time and would be wrong if the response class is changed after `Api.init_app`. I wan't sure about that but was too sloppy to check.

Here's the check:

```py
def outside(obj):
    def inside():
        return obj.attr
    return inside

class Obj:
    def __init__(self, val):
        self.attr = val

o = Obj(12)
f = outside(o)
assert f() == 12
o.attr = 42
assert f() == 42
```

Great. So I can change that to 

```py
            return app.response_class(
                json.dumps(self.specs[app].to_dict(), indent=2),
                mimetype='application/json')
```